sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp

```diff
@@ -3306,10 +3306,20 @@ namespace Azure { namespace Storage { namespace Blobs {
           {
             response.LeaseDuration = BlobLeaseDurationType(x_ms_lease_duration__iterator->second);
           }
-          response.DefaultEncryptionScope
-              = httpResponse.GetHeaders().at("x-ms-default-encryption-scope");
-          response.PreventEncryptionScopeOverride
-              = httpResponse.GetHeaders().at("x-ms-deny-encryption-scope-override") == "true";
+          auto x_ms_default_encryption_scope__iterator
+              = httpResponse.GetHeaders().find("x-ms-default-encryption-scope");
+          if (x_ms_default_encryption_scope__iterator != httpResponse.GetHeaders().end())
+          {
+            response.DefaultEncryptionScope
+                = httpResponse.GetHeaders().at("x-ms-default-encryption-scope");
+          }
+          auto x_ms_deny_encryption_scope_override__iterator
+              = httpResponse.GetHeaders().find("x-ms-deny-encryption-scope-override");
+          if (x_ms_deny_encryption_scope_override__iterator != httpResponse.GetHeaders().end())
+          {
+            response.PreventEncryptionScopeOverride
+                = httpResponse.GetHeaders().at("x-ms-deny-encryption-scope-override") == "true";
+          }
```

Comment:
This code is auto-generated. Please let @Jinming-Hu  to take a look and update the generator if required.